crates/burn-wgpu/src/codegen/dialect/wgsl/compiler.rs

```diff
@@ -147,6 +155,12 @@ impl<F: FloatElement, I: IntElement> Compiler<F, I> {
             gpu::Variable::ConstantScalar(index, elem) => {
                 wgsl::Variable::ConstantScalar(index, Self::compile_elem(elem))
             }
+            gpu::Variable::SharedMemory(index, item, size) => {
+                let item = Self::compile_item(item);
+                self.shared_memories
+                    .push(SharedMemory::new(index, item, size));
```

Comment:
TODO: this declares the shared memory another time each time we use it:
```
var<workgroup> shared_memory_0: array<f32, 1024>;

var<workgroup> shared_memory_0: array<f32, 1024>;

var<workgroup> shared_memory_0: array<f32, 1024>;
```

Reply:
Just a contained index should fix this, since it's supposed to be a very small vector, we shouldn't use a set or an hashmap.